launcher/src/main/java/org/apache/spark/launcher/ChildProcAppHandle.java

```diff
@@ -102,8 +103,20 @@ public synchronized void kill() {
       disconnect();
```

Comment:
I was initially worried that this needs to be in a `try` block but it doesn't look like `disconnect()` is capable of throwing any exceptions.